sherpa/plot/__init__.py

```diff
@@ -2121,11 +2210,13 @@ def plot(self, overplot=False, clearwindow=True, **kwargs):
         Plot.plot(self, self.x, self.y, yerr=self.yerr, xerr=self.xerr,
                   title=self.title, xlabel=self.xlabel, ylabel=self.ylabel,
                   overplot=overplot, clearwindow=clearwindow, **kwargs)
+        super().hline(y=0, xmin=0, xmax=1, linecolor='k',
+                      linewidth=kwargs.get('linewidth', 1), overplot=True)
```

Comment:
For all these residual-style plots, but I didn't want to add it to `delchi` above as there's a separate issue there:

I am not sure I agree with setting the `linewidth` for the `hline` here based on `kwargs`, as I was surprised when

```python

There's also the issue that I was surprised that the horizontal line was affected by

```
plot_resid(linewidth=4)
```

changed the width of the "x axis" line. I've previously mentioned that I believe the axes are drawn at a smaller-than-one scale (something like 0.8), which I would suggest using here. Is there a better way than looking at the axis `spines` attribute?

```
In [19]: for k in plt.gca().spines:
    ...:     print(f"  {k:8s} {plt.gca().spines[k].get_linewidth()}")
    ...: 
  left     0.8
  right    0.8
  bottom   0.8
  top      0.8
```

Reply:
Of course, the trouble is that this mixes up the generic behavior with the backend.

Reply:
Yes, this is one the suggestions that are worthy of discussing. I note that the previous behavior was hardocded at 0.8, because that's the matplotlib default, but would not have matched if I ever made a plot with different axis thickness. How about we just hardcode that 0.8 in here?

There is a trade-off between full customizability for everything and keeping the API clean for users, also reducing the testing / docs / maintenance burden for developers. On second thought, I don't think the line thickness or the x=1 or x=2 line is parameter that needs to be customizable.

Reply:
You'd requested a comment for me on your response, but I don't see something I can respond to here. Or is that you think we should just hard-code to 0.8?

One issue is that it before this logic was in `pylab_backend` whereas here we are talking about the general backend implementation, and I am reluctant to stick a value like 0.8 into `sherpa/plot/__init__.py`.

In the original code this worked because we had xline/ratioline as arguments that got sent to the plot method, and the backend could then decide how best to show them. I haven't thought this through, but can we add an argument to BaseBackend.plot - let's call it yline taking None as it's default argument - and if it is set then the backend draws a line at the yline value. Then where we used to have ratioline you'd set ylnie=1 and for xline you;d set yline=0. My concern is that this then bubbles up and the user will see this, and you seem dead set on not exposing this to users.

Reply:
This is somewhat related to your question at https://github.com/sherpa/sherpa/pull/1382#issuecomment-1156597208

The xaxis and ratioline options actually seem to be more-useful for the code to let the backend take advantage of features the backend has (for instance, ChIPS might have done something fancier than just adding a line - i.e. it could have added an actual axis with tickmarks and labels - but I just checked and it didn't do anyting that fancy).

So I can support either warning if xline or ratioline are set, or just ignoring them, as it's hard to know when to trigger a warning, since technically you can say in CIAO 4.14

```python
plot_data(ratioline=True)
```

Maybe they both default to None and if set to anything we would just warn the user?

I do think, however, that we may internally at least want to keep the behavior of being able to pass to the backend the option of "please draw a line at y=...." in some manner, as otherwise I don't see how we can avoid this issue of backend-specific knowledge slipping into the backend-agnostic code. But I may be too enamoured with how we used to do it to see a better way forward.

Reply:
Backends have the `axhline` and `axvline` functions to draw those lines. That's how I maintain those lines to be drawn in this PR. All that's changing is that I'm taking away the `ratioline` and `xaxis` options to `backend.plot`. Instead the plotting options should just call:
```python
backend.plot(...)
backend.axhline(y=0)
```
I'll add a user warning as you suggested.